### PR TITLE
:sparkles: (go/v3,go/v4-alpha): instead of call script to install kustomize download it first

### PR DESCRIPTION
**Pull request by jberkhahn**
The security management software IBM runs flags this line as a vulnerability and kill -9's it whenever you try to execute it (by running make bundle-build or using any of the operator sdk commands that use that make task). As a result, IBMers currently can't make use of that functionality. This changes up the line a little to be more palatable to  our security software.

### CODE REVIEW

**Comment by jberkhahn** [comment]
/hold
Would like to bring this up for discussion before merging, anyways.

As for the CLA, did that change somehow? I thought the IBM github org was a signatory to the CLA and my membership is public.

**Review by everettraven — APPROVED** [review]
It seems like this would result in the experience for a user being unchanged but not being flagged by automated security alerts. I'm fine with this change going in.

Will wait to add any labels so the bot doesn't auto merge and allow for further discussion

**Comment by jberkhahn** [comment]
todo: backport this to v3

**Review by camilamacedo86 — COMMENTED** [review]
Hi @jberkhahn,

Could you please add in the description the alerts/the output that you are facing which motivates you push this PR?
 
> todo: backport this to v3

You do not backport. You need to change the makefile template for the go/v3 plugin and run make generate to ensure that testdata is updated as well.

**Comment by jberkhahn** [comment]
The security management software IBM runs flags this line as a vulnerability and kill -9's it whenever you try to execute it (by running make bundle-build or using any of the operator sdk commands that use that make task). As a result, IBMers currently can't make use of that functionality. We discussed this a bit internally and at the operator sdk meeting, and we think this is the most light-weight solution. I was going to attend the next kubebuilder meeting and discuss this, but I guess I just missed one so it's not for 2 weeks?

**Comment by camilamacedo86** [comment]
Hi @jberkhahn,

Regards https://github.com/kubernetes-sigs/kubebuilder/pull/3187#issuecomment-1409242059. The meeting is biweekly.  The next one will be on Feb 9. But I do not think we need to bring that to the meeting.

> I was just interested to know what errors were faced. 
> Why download the shell instead of executing directly can sort it out? 

Also, regards the proposed change: could you please update the go/v3 plugin (same boilerplate) and run make generate?

**Comment by jberkhahn** [comment]
Ok, updated this to include v3

**Review by Kavinjsir — COMMENTED** [review]
For the sake of consistency, do we also need to update here:
https://github.com/kubernetes-sigs/kubebuilder/blob/v3.9.0/Makefile#L90

```shell
golangci-lint:
	@[ -f $(GOLANGCI_LINT) ] || { \
	set -e ;\
	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell dirname $(GOLANGCI_LINT)) v1.50.1 ;\
	}
```

**Comment by camilamacedo86** [comment]
Hi @Kavinjsir 

Regards https://github.com/kubernetes-sigs/kubebuilder/pull/3187#pullrequestreview-1278420135

> golangci-lint:
	@[ -f $(GOLANGCI_LINT) ] || { \
	set -e ;\
	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell dirname $(GOLANGCI_LINT)) v1.50.1 ;\
	}

We do not need to update this one because it is ONLY used by those that contributes with Kubebuilder. therefore, it will not affected the end-users.

**Comment by camilamacedo86** [comment]
@jberkhahn I updated the title accordingly.
However, would very great we be able to have in the description of this PR a better info regards the error faced and why the change is required so that others can better understand its motivations by reading that. Could you please provide it?

**Comment by jberkhahn** [comment]
updated the description

**Review by camilamacedo86 — APPROVED** [review]

**Review by varshaprasad96 — APPROVED** [review]
/lgtm

**Pull request merged**
